src/server/search_family.cc

```
@@ -0,0 +1,239 @@
+    }
+
+    return false;
+  }
+
+  DocumentData Serialize() const override {
+    DocumentData out{};
+    for (const auto& [k, v] : *hset_)
+      out[SdsToSafeSv(k)] = SdsToSafeSv(v);
+    return out;
+  }
+
+ private:
+  StringMap* hset_;
+};
+
+unique_ptr<BaseAccessor> GetAccessor(const OpArgs& op_args, const PrimeValue& pv) {
+  if (pv.Encoding() == kEncodingListPack) {
+    auto ptr = reinterpret_cast<ListPackAccessor::LpPtr>(pv.RObjPtr());
+    return make_unique<ListPackAccessor>(ptr);
+  } else {
+    DCHECK_EQ(pv.Encoding(), kEncodingStrMap2);
+    auto* sm = container_utils::GetStringMap(pv, op_args.db_cntx);
+    return make_unique<StringMapAccessor>(sm);
+  }
+}
+
+// Perform brute force search for all hashes in shard with specific prefix
+// that match the query
+void OpSearch(const OpArgs& op_args, string_view prefix, const Query& query,
+              vector<SerializedDocument>* shard_out) {
+  auto& db_slice = op_args.shard->db_slice();
+  DCHECK(db_slice.IsDbValid(op_args.db_cntx.db_index));
+  auto [prime_table, _] = db_slice.GetTables(op_args.db_cntx.db_index);
+
+  string scratch;
+  auto cb = [&](PrimeTable::iterator it) {
+    // Check entry is hash
+    const PrimeValue& pv = it->second;
+    if (pv.ObjType() != OBJ_HASH)
+      return;
+
+    // Check key starts with prefix
+    string_view key = it->first.GetSlice(&scratch);
+    if (key.rfind(prefix, 0) != 0)
+      return;
+
+    // Check entry matches filter
+    auto accessor = GetAccessor(op_args, pv);
+    if (query->Check(search::SearchInput{accessor.get()}))
+      shard_out->emplace_back(key, accessor->Serialize());
+  };
+
+  PrimeTable::Cursor cursor;
+  do {
+    cursor = prime_table->Traverse(cursor, cb);
+  } while (cursor);
+}
+
+}  // namespace
+
+void SearchFamily::FtCreate(CmdArgList args, ConnectionContext* cntx) {
+  string_view idx = ArgS(args, 0);
+  string prefix;
+
+  if (args.size() > 1 && ArgS(args, 1) == "ON") {
+    CHECK_EQ(ArgS(args, 2), "HASH");
```

I'd refrain from using `CHECK*()`s on user-input. Maybe return an error instead?